src/main/java/Siri.java

```diff
@@ -0,0 +1,189 @@
+        System.out.println("    Now you have " + taskList.getTaskCount() + " tasks in the list.");
+    }
+
+    public static void printExitMessage() {
+        System.out.println("    Bye. Hope to see you again soon!");
+        printHorizontalLine();
+    }
+
+    public static void loadTasks() {
+        try {
+            Scanner scanner = SaveFile.getScanner();
+            while (scanner.hasNext()) {
+                String line = scanner.nextLine();
+                String[] data = line.split("\\|");
+                if (data[0].equals("T")) {
+                    taskList.addTask(new Todo(data[2], Boolean.parseBoolean(data[1])));
+                } else if (data[0].equals("D")) {
+                    taskList.addTask(new Deadline(data[2], Boolean.parseBoolean(data[1]), data[3]));
+                } else {
+                    taskList.addTask(new Event(data[2], Boolean.parseBoolean(data[1]), data[3], data[4]));
+                }
+            }
+        } catch (FileNotFoundException e) {
+            try {
+                SaveFile.clearFile();
+            } catch (IOException ex) {
+                System.out.println("Something went wrong: " + e.getMessage());
+            }
+        }
+    }
+
+    public static void saveTasks(TaskList taskList) {
+        try {
+            SaveFile.clearFile();
+            for (Task task: taskList.getTaskList()) {
+                if (task.getClass() == Todo.class) {
+                    SaveFile.appendToFile("T" + "|" +
+                            task.getStatus() + "|" +
+                            task.getDescription() + System.lineSeparator());
+                } else if (task.getClass() == Deadline.class) {
+                    Deadline deadline = (Deadline) task;
+                    SaveFile.appendToFile("D" + "|" +
+                            deadline.getStatus() + "|" +
+                            deadline.getDescription() + "|" +
+                            deadline.getBy() + System.lineSeparator());
+                } else {
+                    Event event = (Event) task;
+                    SaveFile.appendToFile("E" + "|" +
+                            event.getStatus() + "|" +
+                            event.getDescription() + "|" +
+                            event.getFrom() + "|" +
+                            event.getTo() + System.lineSeparator());
+                }
+            }
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+    }
+
+    public static void main(String[] args) {
+        printWelcomeMessage();
+        loadTasks();
+
+        // Request for user input
+        String line = "";
+        Scanner in = new Scanner(System.in);
+
+        // Main loop
+        while (!line.equals("bye")) {
+            line = in.nextLine();
+            printHorizontalLine();
+
+            try {
+                if (line.equals("list")) { // List tasks
```

Comment:
Quite severe case of arrowhead code
One way is to abstract this command processing into a separate method.